utils/project/sync_test.go

```diff
@@ -12,70 +12,188 @@
+			isDir:            true,
+			shouldBeIgnored:  true,
+			ignoredPathsList: []string{},
 		},
 		"success case: directory called not-a-load-test-23498729 should be ignored": {
-			name:            "not-a-load-test-23498729",
-			isDir:           true,
-			shouldBeIgnored: false,
+			name:             "not-a-load-test-23498729",
+			isDir:            true,
+			shouldBeIgnored:  false,
+			ignoredPathsList: []string{},
 		},
 		"success case: directory called noddy_modules should not be ignored": {
-			name:            "noddy_modules",
-			isDir:           true,
-			shouldBeIgnored: false,
+			name:             "noddy_modules",
+			isDir:            true,
+			shouldBeIgnored:  false,
+			ignoredPathsList: []string{},
 		},
 		"success case: file called .DS_Store should be ignored": {
-			name:            ".DS_Store",
-			isDir:           false,
-			shouldBeIgnored: true,
+			name:             ".DS_Store",
+			isDir:            false,
+			shouldBeIgnored:  true,
+			ignoredPathsList: []string{},
 		},
 		"success case: file called something.swp should be ignored": {
-			name:            "something.swp",
-			isDir:           false,
-			shouldBeIgnored: true,
+			name:             "something.swp",
+			isDir:            false,
+			shouldBeIgnored:  true,
+			ignoredPathsList: []string{},
 		},
 		"success case: file called something.swpnot should not be ignored": {
-			name:            "something.swpnot",
-			isDir:           false,
-			shouldBeIgnored: false,
+			name:             "something.swpnot",
+			isDir:            false,
+			shouldBeIgnored:  false,
+			ignoredPathsList: []string{},
 		},
 		"success case: file called node_modules should not be ignored": {
-			name:            "node_modules",
-			isDir:           false,
-			shouldBeIgnored: false,
+			name:             "node_modules",
+			isDir:            false,
+			shouldBeIgnored:  false,
+			ignoredPathsList: []string{},
 		},
 		"success case: directory called .DS_Store should not be ignored": {
-			name:            ".DS_Store",
-			isDir:           true,
-			shouldBeIgnored: false,
+			name:             ".DS_Store",
+			isDir:            true,
+			shouldBeIgnored:  false,
+			ignoredPathsList: []string{},
+		},
+		"success case: path containing noddy_modules should be ignored as it is in .cw-settings": {
+			name:             "noddy_modules",
+			isDir:            true,
+			shouldBeIgnored:  true,
+			ignoredPathsList: []string{"noddy_modules"},
 		},
 	}
 	for name, test := range tests {
 		t.Run(name, func(t *testing.T) {
-			fileIsIgnored := ignoreFileOrDirectory(test.name, test.isDir)
+			fileIsIgnored := ignoreFileOrDirectory(test.name, test.isDir, test.ignoredPathsList)
 
 			assert.IsType(t, test.shouldBeIgnored, fileIsIgnored, "Got: %s", fileIsIgnored)
 
 			assert.Equal(t, test.shouldBeIgnored, fileIsIgnored, "fileIsIgnored was %b but should have been %b", fileIsIgnored, test.shouldBeIgnored)
 		})
 	}
 }
+
+func TestRetrieveIgnoredPathsListFromCWSettings(t *testing.T) {
```

Comment:
If you change the func name above, this will also change too

Reply:
Changed